modules/oe_subscriptions_anonymous/tests/src/FunctionalJavascript/SubscribeModalFormTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_subscriptions_anonymous\Functional;
+
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+
+/**
+ * Modal form test.
+ */
+class SubscribeModalFormTest extends WebDriverTestBase {
+
+  use FlagCreateTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'node',
+    'flag',
+    'oe_subscriptions',
+    'oe_subscriptions_anonymous',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Tests that subscribe link open a modal.
+   */
+  public function testModalForm(): void {
+    // Create an article content type.
+    $this->drupalCreateContentType([
+      'type' => 'article',
+      'name' => 'Article',
+    ]);
+    // Create a flag.
+    $flag = $this->createFlagFromArray([
+      'id' => 'subscribe_article',
+      'short_text' => 'Subscribe article',
+      'entity_type' => 'node',
+      'bundles' => ['article'],
+    ]);
+    // Create the node.
+    $node = $this->drupalCreateNode([
+      'type' => 'article',
+      'status' => 1,
+    ]);
+    $node->save();
+
+    $page = $this->getSession()->getPage();
+    $assert_session = $this->assertSession();
+    // Using ui-dialog parent of modal-form,
+    // Submit buttons and close in siblings.
```

Comment:
I am not sure I understand this.

Reply:
The form has ui-dialog as parent, and then modal-fom and ui-dialog-buttonpane as children at the same level, just wanted to clarify I couldn't use modal-form.